runtime/pangolin/src/pallets/governance/track.rs

```diff
@@ -0,0 +1,236 @@
+			min_approval: pallet_referenda::Curve::make_reciprocal(
+				1,
+				2,
+				percent(96),
+				percent(50),
+				percent(100),
+			),
+			min_support: pallet_referenda::Curve::make_reciprocal(
+				1,
+				2,
+				percent(1),
+				percent(0),
+				percent(2),
+			),
+		},
+	),
+	(
+		2,
+		pallet_referenda::TrackInfo {
+			name: "general_admin",
+			max_deciding: 10,
+			decision_deposit: DARWINIA_PROPOSAL_REQUIREMENT,
+			prepare_period: 2 * MINUTES,
+			decision_period: 2 * MINUTES,
+			confirm_period: 2 * MINUTES,
+			min_enactment_period: 2 * MINUTES,
+			min_approval: pallet_referenda::Curve::make_reciprocal(
+				1,
+				2,
+				percent(80),
+				percent(50),
+				percent(100),
+			),
+			min_support: pallet_referenda::Curve::make_reciprocal(
+				1,
+				2,
+				percent(10),
+				percent(0),
+				percent(50),
+			),
+		},
+	),
+	(
+		3,
+		pallet_referenda::TrackInfo {
+			name: "referendum_canceller",
+			max_deciding: 20,
+			decision_deposit: DARWINIA_PROPOSAL_REQUIREMENT,
+			prepare_period: 2 * MINUTES,
+			decision_period: 2 * MINUTES,
+			confirm_period: 2 * MINUTES,
+			min_enactment_period: 2 * MINUTES,
+			min_approval: pallet_referenda::Curve::make_reciprocal(
+				1,
+				2,
+				percent(96),
+				percent(50),
+				percent(100),
+			),
+			min_support: pallet_referenda::Curve::make_reciprocal(
+				1,
+				2,
+				percent(1),
+				percent(0),
+				percent(10),
+			),
+		},
+	),
+	(
+		4,
+		pallet_referenda::TrackInfo {
+			name: "referendum_killer",
+			max_deciding: 100,
+			decision_deposit: DARWINIA_PROPOSAL_REQUIREMENT,
+			prepare_period: 2 * MINUTES,
+			decision_period: 2 * MINUTES,
+			confirm_period: 2 * MINUTES,
+			min_enactment_period: 2 * MINUTES,
+			min_approval: pallet_referenda::Curve::make_reciprocal(
+				1,
+				2,
+				percent(96),
+				percent(50),
+				percent(100),
+			),
+			min_support: pallet_referenda::Curve::make_reciprocal(
+				1,
+				2,
+				percent(1),
+				percent(0),
+				percent(10),
+			),
+		},
+	),
+];
+
+pub struct TracksInfo;
+impl pallet_referenda::TracksInfo<Balance, BlockNumber> for TracksInfo {
+	type Id = u16;
+	type RuntimeOrigin = <RuntimeOrigin as frame_support::traits::OriginTrait>::PalletsOrigin;
+
+	fn tracks() -> &'static [(Self::Id, pallet_referenda::TrackInfo<Balance, BlockNumber>)] {
+		&TRACKS_DATA[..]
+	}
+
+	fn track_for(id: &Self::RuntimeOrigin) -> Result<Self::Id, ()> {
+		if let Ok(system_origin) = frame_system::RawOrigin::try_from(id.clone()) {
+			match system_origin {
+				frame_system::RawOrigin::Root => {
+					if let Some((track_id, _)) =
+						Self::tracks().iter().find(|(_, track)| track.name == "root")
+					{
+						Ok(*track_id)
+					} else {
+						Err(())
+					}
+				},
+				_ => Err(()),
+			}
+		} else if let Ok(custom_origin) = custom_origins::Origin::try_from(id.clone()) {
+			if let Some((track_id, _)) = Self::tracks().iter().find(|(_, track)| {
+				if let Ok(track_custom_origin) = custom_origins::Origin::from_str(track.name) {
```

Comment:
If the input is `GeneralAdmin` custom origin, no track_id can be matched following the logic here.